nvm.sh

```diff
@@ -1848,13 +1848,14 @@ nvm_install_source() {
 
   local make
   make='make'
-  if [ "${NVM_OS}" = 'freebsd' ]; then
-    make='gmake'
-    MAKE_CXX='CXX=c++'
-  elif [ "${NVM_OS}" = 'aix' ]; then
+  if [ "${NVM_OS}" = 'freebsd' ] || [ "${NVM_OS}" = 'aix' ]; then
     make='gmake'
   fi
-  if nvm_has "clang++" && nvm_has "clang" && nvm_version_greater_than_or_equal_to nvm_clang_version 3.5 ; then
+  if [ "${NVM_OS}" = 'freebsd' ] || [ "${NVM_OS}" = 'darwin' ]; then
+    [ -z "${CXX-}" ] && MAKE_CXX='CC=cc CXX=c++'
```

Comment:
We ONLY set it for FreeBSD before, I don't know how many FreeBSD users are using nvm, but I guess most of the users is on Linux distros, and then MacOS, no CXX, CC were set in this two situations, so it makes sense there is no issue about it.

If we'll always overwrite CC and CXX, then export CC and CXX in `.travis.yml` makes no sense. For distros like Ubuntu 12.04/14.04 doesn't have clang new enough, install clang from ppa or llvm official source will make the newer versions called clang-3.x, then nvm can't not detect it, after all, what's the reason why we should not respect the environment variable? Do we really got a problem that the incorrect CXX and CC environment variable set made the build failed?